Engine/tracker.py

```diff
@@ -5,209 +5,203 @@
-
-        self._ball.predict()
+            
+        for ball in self._balls:
+            ball.predict()
 
     def remove_undetected(self):
-        active_robots = (robot for robot in self._yellow_team + self._blue_team if robot.is_active)
-        for robot in active_robots:
-            if self._current_timestamp - robot.last_capture_time > Tracker.MAX_UNDETECTED_DELAY:
-                robot.reset()
-                self.logger.debug('Robot %d was undetected for more than %d seconds.',
-                                  robot.id,
-                                  Tracker.MAX_UNDETECTED_DELAY)
-
-        if self._ball.is_active:
-            if self._current_timestamp - self._ball.last_capture_time > Tracker.MAX_UNDETECTED_DELAY:
-                self._ball.reset()
-                self.logger.debug('The ball was undetected for more than %d seconds.',
-                                  Tracker.MAX_UNDETECTED_DELAY)
+
+        for team_color, robots in self.active_robots.items():
+            for robot in robots:
+                if self.timestamp - robot.last_update_time > config['ENGINE']['max_undetected_robot_time']:
+                    robot.reset()
+                    self.logger.debug('Robot %d of %s team was undetected for more than %d seconds.',
+                                      robot.id,
+                                      team_color,
+                                      config['ENGINE']['max_undetected_robot_time'])
+
+        for ball in self.active_balls:
+            if self.timestamp - ball.last_update_time > config['ENGINE']['max_undetected_ball_time']:
+                ball.reset()
+                self.logger.debug('Ball %d was undetected for more than %d seconds.',
+                                  ball.id,
+                                  config['ENGINE']['max_undetected_ball_time'])
 
     @staticmethod
     def _put_in_world_referential(robot, cmd):
-        if Config()["GAME"]["on_negative_side"]:
+        if config['GAME']['on_negative_side']:
             cmd.position = rotate(cmd.position, -np.pi - robot.orientation)
             cmd.x *= -1
             cmd.orientation *= -1
         else:
             cmd.position = rotate(cmd.position, robot.orientation)
         return cmd
 
-
     @staticmethod
-    def change_reference(detection_frame: Dict[str, List[Dict[str, Any]]]) -> Dict[str, List[Dict[str, Any]]]:
+    def _change_frame_side(detection_frame: Dict[str, List[Dict[str, Any]]]) -> Dict[str, List[Dict[str, Any]]]:
 
-        teams = detection_frame.get('robots_blue', []) \
-              + detection_frame.get('robots_yellow', [])
-
-        for robot_obs in teams:
+        for robot_obs in detection_frame.get('robots_blue', []) + detection_frame.get('robots_yellow', []):
             robot_obs['x'] *= -1
-            robot_obs['orientation'] = wrap_to_pi(np.pi - robot_obs['orientation'])
+            robot_obs['orientation'] *= -1
```

Comment:
We need to mirror the robot orientation around the y axis. Not around the x axis. For example a robot with a 110 deg orientation has for mirror 70 deg, not -110 deg

Reply:
oh, i copied pasted the other computation...

Reply:
oh... ust realize it's not the same thing at all lawl sorry